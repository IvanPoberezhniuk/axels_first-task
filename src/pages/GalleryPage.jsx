import React, {useEffect, useState} from 'react';
import {Col, Container, Row} from 'react-bootstrap';
import {useDispatch, useSelector} from 'react-redux';

import {ImageCard, ImageModal} from '../components';

import {Link, useLocation} from 'react-router-dom';
import {fetchImages} from '../store/modules/images';

const GalleryPage = () => {
  const [modalShow, setModalShow] = useState(false);
  const [imgId, setImgId] = useState(null);
  const location = useLocation();
  //redux
  const dispatch = useDispatch();
  const imagesList = useSelector(store => store.imagesStore.images);

  useEffect(() => {
    dispatch(fetchImages());
  }, []);

  const setModal = (id) => {
    setModalShow(true);
    setImgId(id);
  };

  return (
    <div>
      <Container>
        <Row>
          {imagesList.map((img) => (
            <Col xs={12} md={6} lg={4} xl={4} key={img.id}>
              <Link to={{pathname: `/gallery/${img.id}`, state: {background: location}}}>
                <ImageCard url={img.url} showModal={() => setModal(img.id)}/>
              </Link>
            </Col>)
          )}
        </Row>
      </Container>
      {imgId && <ImageModal show={modalShow} imgId={imgId} onHide={() => {
        setModalShow(false);
      }}/>}
    </div>
  );
};

GalleryPage.propTypes = {};

export default GalleryPage;
