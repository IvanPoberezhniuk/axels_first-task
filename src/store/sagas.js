import {all} from '@redux-saga/core/effects';
import {watcherFetchImages, watcherFetchImageDetails, watcherAddComment,} from './modules/images';

export default function* rootSaga() {
  yield all([
    watcherFetchImages(),
    watcherFetchImageDetails(),
    watcherAddComment()
  ]);
}
