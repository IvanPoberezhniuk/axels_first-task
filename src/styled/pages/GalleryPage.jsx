import React, {useEffect, useState} from 'react';
import ImageCard from '../components/ImageCard';
import {getImages} from '../api/httpRequests';
import {Col, Container, Row} from 'react-bootstrap';
import ImageModal from '../components/ImageModal';

const GalleryPage = () => {
  const [imgList, setImgList] = useState([]);
  const [modalShow, setModalShow] = useState(false);
  const [imgId, setImgId] = useState(null);

  useEffect(() => {
    getImages().then(data => setImgList(data));
  }, []);

  const setModal = (imgId) => {
    setModalShow(true);
    setImgId(imgId);
  };

  return (
    <div>
      <Container>
        <Row>
          {imgList.map((img) => (
            <Col xs={12} md={6} lg={4} key={img.id}>
              <ImageCard url={img.url} showModal={() => setModal(img.id)}/>
            </Col>)
          )}
        </Row>
      </Container>
      {imgId && <ImageModal show={modalShow} imgId={imgId} onHide={() => setModalShow(false)}/>}
    </div>
  );
};

GalleryPage.propTypes = {};

export default GalleryPage;
