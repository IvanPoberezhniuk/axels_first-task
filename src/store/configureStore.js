import {configureStore, getDefaultMiddleware} from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';

import imagesReducer from './modules/images';
import createLogger from 'redux-logger';
import sagas from './sagas';

const sagaMiddleware = createSagaMiddleware();

const middleware = [...getDefaultMiddleware({thunk: false}), sagaMiddleware];

const store = () => {
  const store = configureStore({
    reducer: {
      imagesStore: imagesReducer,
    },
    middleware
  });
  sagaMiddleware.run(sagas);

  return store;
};

export default store();

