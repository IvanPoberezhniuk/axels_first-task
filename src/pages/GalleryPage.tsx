import React, { useEffect } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { RootStateOrAny, useDispatch, useSelector } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';

import { ImageCard } from '../components';

import { fetchImages, Image } from '../redux/ducks/images';
import { AppDispatch } from '../redux/configureStore';

const GalleryPage = () => {
  const location = useLocation();
  const dispatch: AppDispatch = useDispatch();
  const imagesList: Array<Image> = useSelector((store: RootStateOrAny) => store.imagesStore.images);

  useEffect(() => {
    dispatch(fetchImages());
  }, [dispatch]);

  return (
    <div>
      <Container>
        <Row>
          {imagesList.map((img: Image) => (
            <Col xs={12} md={6} lg={4} xl={4} key={img.id}>
              <Link
                to={{
                  pathname: `/gallery/${img.id}`,
                  state: { background: location }
                }}
              >
                <ImageCard url={img.url} />
              </Link>
            </Col>
          ))}
        </Row>
      </Container>
    </div>
  );
};

export default GalleryPage;
