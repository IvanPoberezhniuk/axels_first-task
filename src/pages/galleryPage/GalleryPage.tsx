import React, { useEffect } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';
import { ImageCard } from '../../components';

import { fetchImages, Image } from '../../redux/ducks/images';
import { AppDispatch } from '../../redux/configureStore';
import Selectors from '../../redux/selectors';

const GalleryPage = (props: any) => {
  const location = useLocation();
  const dispatch: AppDispatch = useDispatch();
  const imagesList: Array<Image> = useSelector(Selectors.images);

  useEffect(() => {
    dispatch(fetchImages());
  }, [dispatch, props]);

  return (
    <div>
      <Container>
        <Row>
          {imagesList && imagesList.map((img: Image) => (
            <Col xs={12} md={6} lg={4} xl={4} key={img.id}>
              <Link
                to={{
                  pathname: `/gallery/${img.id}`,
                  state: { background: location }
                }}
              >
                <ImageCard url={img.url} />
              </Link>
            </Col>
          ))}
        </Row>
      </Container>
    </div>
  );
};

export default GalleryPage;
