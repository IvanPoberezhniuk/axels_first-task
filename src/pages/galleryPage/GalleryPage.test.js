import * as redux from 'react-redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import GalleryPage from './GalleryPage';
import store from '../../redux/configureStore';
import React from 'react';


let component;
const history = { pathname: '/' };
const mockHook = jest.fn();

const setUp = (props) => mount(
  <Provider store={store}>
    <MemoryRouter history={history}>
      <GalleryPage {...props} />
    </MemoryRouter>
  </Provider>);

const mockedUseEffect = jest.spyOn(React, 'useEffect').mockImplementation(mockHook);
const mockedUseDispatch = jest.spyOn(redux, 'useDispatch').mockImplementation(mockHook);

// jest.mock('react', () => ({
//   ...jest.requireActual('react'),
//   useLayoutEffect: jest.requireActual('react').useEffect
// }));

describe('GalleryPage component Snapshot', () => {
  it('SNAPSHOT', () => {
    expect(component).toMatchSnapshot();
  });
  it('useEffect ToBeCalled 1 time', () => {
    expect(mockedUseEffect).toBeCalledTimes(1);
    expect(mockedUseDispatch).toBeCalledTimes(1);
  });
});
