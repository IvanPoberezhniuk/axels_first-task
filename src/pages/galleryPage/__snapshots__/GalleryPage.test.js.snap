// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPage component Snapshot SNAPSHOT 1`] = `undefined`;
